tests/ecmascript/test-bi-nodejs-buffer-tostring.js

```diff
@@ -241,20 +240,21 @@ function nodejsBufferToStringTest() {
     // buf.toString([encoding], [start], [end])
 
     // Without arguments encoding defaults to UTF-8 and the entire
-    // buffer is converted to string.  At least undefined and null
+    // buffer is converted to string.  At least undefined
     // are accepted as "not defined" for encoding.
     b = new Buffer('ABC');
     safePrintString(b.toString());
     safePrintString(b.toString(undefined));
-    safePrintString(b.toString(null));
+    // null is not a valid encoding
+    try { safePrintString(b.toString(null)); } catch(e) { }
```

Comment:
This covers the `null` argument, but the catch clause should probably print something like `print(e.name)` (i.e. TypeError) so it's covered in the expected output explicitly.

Also there should now be explicit coverage for these encodings at least:
- `utf8`
- `utf-8`
- `wtf`

Node.js seems to accept both `utf8` and `utf-8` so it'd be nice to cover that behavior explicitly, whatever the current behavior is.

Reply:
More tests were added here and the relevant tests in `Buffer.isEncoding` were updated